cmd/controller/main.go

```diff
@@ -43,6 +47,20 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	client := op.Manager.GetAPIReader()
```

Comment:
Can we consider scoping this into a function to make it easier to grok what this is validating, something like `ValidateEC2NodeClassAMISelectorTerms`?